### ISSUE TITLE
Please allow creating an agent service without enabling internalTrafficPolicy

### ISSUE THREAD

**Issue by mbyio** [DataDog/helm-charts#612]
**Describe what happened:**

We use Kuma Mesh (https://kuma.io/). Kuma's sidecars can only send traces to a service, not to the agent running on the same node. So, to send traces to datadog, we have to create our own kubernetes service, which selects datadog daemonset pods.

**Describe what you expected:**

I would like the datadog helm chart to be able to create that service for us. Actually, it already can, but only if you enable "internalTrafficPolicy". In the long-term, we would like to turn that on, but it is still a beta feature in Kubernetes, and we won't be able to support it in our clusters for years.

So basically I'd like you to add an if statement around https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/agent-services.yaml#L95 so we can create the service without it.

**Steps to reproduce the issue:**

**Additional environment details (Operating System, Cloud provider, etc):**

**Comment by clamoriniere** [comment]
Hi @mbyio 
Thanks for opening this issue. Your problem description is very clear.

Indeed the `internalTrafficPolicy` is only available in beta from Kubernetes `1.21` and in alpha in `1.20`.

Let me explain why we only create the service if `internalTrafficPolicy` is available on a cluster. The Datadog agent allows to tag automatically any observability signals of you application running on the same host. To do so, the agent needs to access several APIs/Interfaces on the host (kubelet, cgroup...). The extracted information are only available locally on the agent. To have the applicative traces properly tagged, the data need to be send on the agent running on the same host, that is why  `internalTrafficPolicy` is useful in that case.

If we were creating a "classic" service, the data send by your applicative pods will be dispatch to any Datadog agent pods on the cluster with the standard round robin algorithm, which break tagging of the trace.

does Kuma support hostPort, hostNetwork? can you exclude port or ip range from the Kuma service mesh?

**Comment by mbyio** [comment]
Thank you @clamoriniere! I suspected that you collected more information if I send to the same host, but wasn't sure, thanks for confirming it. I am mostly concerned about the potential performance impact myself.

Unfortunately Kuma only lets you provide a URL to send the traces to. Here's the relevant documentation with an example for Datadog: https://kuma.io/docs/1.5.x/policies/traffic-trace/#set-up-in-kuma So I haven't thought of a way to use the URL to make kuma talk to the same host. We also have an open issue with them to add support for the ability to send to the host IP: https://github.com/kumahq/kuma/issues/3732

**Comment by jacjtito** [comment]
I'm closing this stale issue. Feel free to reopen it or create a new issue if needed.
